src/components/InstrumentPicker/InstrumentList.tsx

```diff
@@ -41,9 +41,22 @@ class InstrumentList extends React.Component<
       this.state.sortByParam === sortByParam
         ? sortDirection
         : SortDirection.DESC;
-    this.setState(
-      sortData(this.props.instruments, sortByParam, direction, this.state)
+    const shouldBeSorted = this.props.instruments.filter(i => !!i[sortByParam]);
```

Comment:
instrumentsToBeSorted
variables started from "should" looks more like boolean type